test/integration-test.py

```diff
@@ -168,19 +169,32 @@ def verify_ct_submission(expectedSubmissions, url):
         die(ExitStatus.CTFailure)
     return 0
 
-def run_node_test(domain, chall_type, expected_ct_submissions):
+def run_node_test(domain, chall_type, expected_ct_submissions, abort_step="null"):
     email_addr = "js.integration.test@letsencrypt.org"
     cert_file = os.path.join(tempdir, "cert.der")
     cert_file_pem = os.path.join(tempdir, "cert.pem")
     key_file = os.path.join(tempdir, "key.pem")
+
+    if abort_step is None or abort_step == "":
+        abort_step = "null"
+
     # Issue the certificate and transform it from DER-encoded to PEM-encoded.
-    if subprocess.Popen('''
+    exit_code = subprocess.Popen('''
         node test.js --email %s --agree true \
           --domains %s --new-reg http://localhost:4000/acme/new-reg \
-          --certKey %s --cert %s --challType %s && \
+          --certKey %s --cert %s --challType %s --abort-step %s && \
         openssl x509 -in %s -out %s -inform der -outform pem
-        ''' % (email_addr, domain, key_file, cert_file, chall_type, cert_file, cert_file_pem),
-        shell=True).wait() != 0:
+        ''' % (email_addr, domain, key_file, cert_file, chall_type, abort_step, cert_file, cert_file_pem),
+                                 shell=True).wait()
+
+    if abort_step != "null":
+        if exit_code == ABORT_SUCCESS:
+            return ABORT_SUCCESS
+        else:
+            print("\nDid not properly abort at requested phase")
```

Comment:
I think the extra complexity required to check the aborted status is unnecessary. Test.js should just exit successfully when it hits the abortstep.